python_modules/libraries/dagster-components/dagster_components/core/component_decl_builder.py

```diff
@@ -39,11 +57,36 @@ def from_path(component_file_path: Path) -> "YamlComponentDecl":
         )
 
         return YamlComponentDecl(
-            path=component_file_path.parent,
+            path=path,
             component_file_model=obj,
             source_position_tree=parsed.source_position_tree,
         )
 
+    def get_component_type(self, registry: ComponentTypeRegistry) -> type[Component]:
```

Comment:
nit: `get_` belies that this is doing some real work (parsing python files), maybe `resolve_component_type`?

Reply:
might be worth a brief docstr to explain that this does local lookup in addition to the expected registry lookup